### PR TITLE
Jest ci setup

### PR DESCRIPTION
**Pull request by Naseem77**
## Summary by CodeRabbit

- **New Features**
	- Enhanced CI workflow for FalkorDB with improved testing capabilities.
	- Introduced a new asynchronous function for establishing database connections.
	- New test suite for validating graph operations, including creation, querying, and error handling.
	- Added a utility function to generate random numbers for testing.
	- Configured Jest for TypeScript testing, streamlining the testing process.

- **Bug Fixes**
	- Improved error handling during database connection attempts.

- **Tests**
	- Comprehensive tests added for various functionalities of the FalkorDB client, including graph creation and query execution.

### CODE REVIEW

File: .github/workflows/jest.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+name: Run Jest Tests
+
+on:
```
Comment (AviAvni) [inline .github/workflows/jest.yml:3]: can you migrate this to the already exsited action in nodes.yml

**Review by AviAvni — CHANGES_REQUESTED** [review]
please also look at the [coderabbitai](https://github.com/apps/coderabbitai) comments

File: package.json [hunk 2]
```diff
@@ -47,6 +48,7 @@
   "dependencies": {
     "@redis/client": "^1.6.0",
     "cluster-key-slot": "1.1.2",
+    "falkordb": "^6.2.2",
```
[the remark this answers is not in the document]
Comment (Naseem77) [reply package.json:51]: shortcut to run jest tests: npm run jest

File: .github/workflows/jest.yml (hunk 1 above)
Comment (Naseem77) [reply .github/workflows/jest.yml]: Done

**Review by AviAvni — APPROVED** [review]

**Comment by Naseem77** [comment]
This pull request addresses and resolves the following issues: #161, #162, #163, #164, and #165

**Pull request merged**